ios/Handlers/RNPanHandler.m

```diff
@@ -84,9 +86,10 @@ - (void)touchesMoved:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
     self.state = UIGestureRecognizerStateFailed;
     return;
   }
-  if ((self.state == UIGestureRecognizerStatePossible || self.state == UIGestureRecognizerStateChanged) && _gestureHandler.shouldCancelWhenOutside) {
+  if ((self.state == UIGestureRecognizerStatePossible || self.state == UIGestureRecognizerStateChanged)) {
     CGPoint pt = [self locationInView:self.view];
-    if (!CGRectContainsPoint(self.view.bounds, pt)) {
+    _isWithinBounds = [_gestureHandler containsPointInView:pt];
```

Comment:
can we calculate `isWithinBounds` on demand instead of storing it under _isWithinBounds variable? It also seem that `containsPointInView` method you added would be a good candidate for that. It also seem like it does not need to take `pt` as argument because you always call to `[self locationInView:]` prior to using it